admin/server/deployment.go

```diff
@@ -762,17 +806,82 @@ func (s *Server) GetIFrame(ctx context.Context, req *adminv1.GetIFrameRequest) (
 	}, nil
 }
 
-// getAttributesFor returns a map of attributes for a given user and project.
+// GetResourceRestrictionsForUser returns resource restrictions for a given user and project.
+func (s *Server) GetResourceRestrictionsForUser(ctx context.Context, projID, userID string) (bool, []database.ResourceName, error) {
+	mu, err := s.admin.DB.FindProjectMemberUser(ctx, projID, userID)
+	if err != nil && !errors.Is(err, database.ErrNotFound) {
+		return false, nil, err
+	}
+	mug, err := s.admin.DB.FindProjectMemberUsergroupsForUser(ctx, projID, userID)
+	if err != nil {
+		return false, nil, err
+	}
+	// filter out managed usergroups as they do not have resource restrictions
+	var filteredMug []*database.MemberUsergroup
+	for _, g := range mug {
+		if !g.Managed {
+			filteredMug = append(filteredMug, g)
+		}
+	}
```

Comment:
Why would they not have that? For example, this seems like a real use case to me:
```
rill usergroup set-role --group autogroup:guests --explore external_dashboard
```

Reply:
Since resource restrictions are on project level, we check only usergroups having a project role, which mean it will either be a user created group or `autogroup:member` I think. So I think we should not allow restricting resources on autorgoups may be as it will cause confusion? 

If we want then usergroup `autogroup:members` having role on each project should have default value for `restrict_resources` as `true` instead of `false` so that ANDing with other usergroups/direct membership works correctly otherwise it will always return false and no resource restriction will happen.